src/main/java/ultoi/command/MarkCommand.java

```diff
@@ -0,0 +1,82 @@
+package ultoi.command;
+
+import ultoi.task.Task;
+import ultoi.task.ToDo;
+import ultoi.task.Deadline;
+import ultoi.task.Event;
+
+import ultoi.util.Ultoi;
+import ultoi.util.UltoiUi;
+import ultoi.util.UltoiException;
+import ultoi.util.Storage;
+import ultoi.util.TaskList;
+import ultoi.util.Parser;
+import ultoi.util.DateTime;
+
+/**
+ * Represents a command that marks a task as done or undone.
+ *
+ * @author snoidetx
+ * @version 0.0.0
+ */
+public class MarkCommand implements Command {
+    private static final int COMMAND_LENGTH_MARK = 4;
+    private static final int COMMAND_LENGTH_UNMARK = 6;
+    private static final String MESSAGE_MARK = "Nice! I have marked this task as done:";
+    private static final String MESSAGE_UNMARK = "Okay! I have marked this task as not done yet:";
+
+    private final boolean isMark;
+    private final int index;
+
+    /**
+     * Creates an mark command.
+     *
+     * @param input User input.
+     * @throws UltoiException If the input cannot be identified.
+     */
+    public MarkCommand(String input) throws UltoiException {
+        this.isMark = input.startsWith("mark");
```

Comment:
Clever way to combine mark and unmark commands into one class by making use of the boolean isMark.